controller_routing.go

```diff
@@ -61,8 +61,8 @@ func (c *controller) manageData() {
 				}
 			case TypePeerResponse:
 				c.shareMtx.RLock()
-				if f, ok := c.shareListener[peer.NodeID]; ok {
-					f(parcel)
+				if c, ok := c.shareListener[peer.NodeID]; ok {
```

Comment:
The `c` overrides the existing "c" controller variable. It has no negative effect in this case but please rename it to `ch` or `async`.